pkg/provider/provider.go

```diff
@@ -621,3 +636,83 @@ func validateFileName(fileName string) error {
 	}
 	return nil
 }
+
+type node struct {
+	cert     *x509.Certificate
+	parent   *node
+	isParent bool
+}
+
+func fetchCertChains(data []byte) ([]byte, error) {
+	var newCertChain []*x509.Certificate
+	var pemData []byte
+	nodes := make([]*node, 0)
+
+	for {
+		// decode pem to der first
+		block, rest := pem.Decode(data)
+		data = rest
+
+		if block == nil {
+			break
+		}
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
```

Comment:
maybe also explicitly check for nil cert

Reply:
Added the nil cert check.